src/Umbraco.Web/Cache/DistributedCacheExtensions.cs

```diff
@@ -189,16 +190,16 @@ public static void RefreshAllPublishedSnapshot(this DistributedCache dc)
 
         #region MacroCache
 
-        public static void RefreshMacroCache(this DistributedCache dc, IMacro macro)
+        public static void RefreshMacroCache(this DistributedCache dc, IJsonSerializer jsonSerializer, IMacro macro)
```

Comment:
We shouldn't pass in a serializer here (see prev comments). The `MacroCacheRefresher.Serialize` shouldn't be static, instead we should resolve the cache refresher instance from the collection and call a non-static Serialize method on it. Same with the RemoveMacroCache method. 

Should be the same for all json cache refreshers, we should remove their static methods and require resolving the instance to access it's serialize/deserialize methods and the cache refresher has it's own injected serializer to do the work.